### DISCUSSION TITLE
addon: code of the file

### DISCUSSION THREAD

**Discussion by sajcics** [storybookjs/storybook#14004] [Help]
**Describe the bug**
Storybook addon is not showing right code snippet of the story. When I have multiple stories for one component, and export stories in one file, storybook is not recognizing when should show correct code snippets. 
For example I have a file `Component1.stories.js`:
``` 
export * from './first.stories.js';
export * from './second.stories.js;

export default {
 title: 'Component1',
 component: 'Component1'
}
``` 
and inside storybook if I go to `First` story or `Second` story it will show code from `Component1.stories.js` event it should recognize when I'm in specific story. Btw, `First.stories.js` and `Second.stories.js` are not export any default values such as `Component1.stories.js`, which means that I am exporting those file correctly inside of `Component1.stories.js`. 

**To Reproduce**

1. Make several story files
2. export all in one file
3. Open storybook and go to one of those seperated stories
4. Open storybook addon and see which code is shown for that story

**Expected behavior**
Storybook addon for code snippets should recognize correct file which should show code. 
If I have file structure in storybook like this:
<ul>
 <li>Component1</li>
 <li>First</li>
 <li>Second</li>
</ul>
then opening First story should show code related to that story.

**System**
```
  System:
    OS: Linux 5.4 Ubuntu 18.04.5 LTS (Bionic Beaver)
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  Binaries:
    Node: 10.19.0 - ~/.nvm/versions/node/v10.19.0/bin/node
    Yarn: 1.22.10 - ~/.nvm/versions/node/v10.19.0/bin/yarn
    npm: 6.13.4 - ~/.nvm/versions/node/v10.19.0/bin/npm
  Browsers:
    Firefox: 84.0.2
  npmPackages:
    @storybook/addon-actions: ^6.1.14 => 6.1.14 
    @storybook/addon-backgrounds: ^6.1.14 => 6.1.14 
    @storybook/addon-docs: ^6.1.14 => 6.1.14 
    @storybook/addon-knobs: ^6.1.14 => 6.1.14 
    @storybook/addon-links: ^6.1.14 => 6.1.14 
    @storybook/addon-storyshots: ^6.1.14 => 6.1.14 
    @storybook/addon-storyshots-puppeteer: ^6.1.14 => 6.1.14 
    @storybook/addon-storysource: ^6.1.14 => 6.1.14 
    @storybook/addon-viewport: ^6.1.14 => 6.1.14 
    @storybook/addons: ^6.1.14 => 6.1.14 
    @storybook/preset-create-react-app: ^3.1.5 => 3.1.5 
    @storybook/react: 6.1.14 => 6.1.14 ```

**Comment by crixx** [comment]
hi @sajcics 

I don't exactly understand what you really try to achieve. If I understand correctly, you want to have a the same as if you had a single file that looks like this below, but want to split those "FirstStory" "SecondStory" into single files?

````
export default {
 title: 'Component1',
 component: 'Component1'
}

export const FirstStory = (args) => <YourComponent {...args} />;

export const SecondStory = (args) => <YourComponent {...args} />;
````

**Comment by shilman** [comment]
are you looking for this? https://www.npmjs.com/package/@storybook/addon-storysource
